### PR TITLE
#169817862:user signup

### PR DESCRIPTION
**Pull request by mujohn26**
#### What does this PR do?

-  user should be able to create account

#### Description of Task to be completed?

-  user must type the fields to create
-  user will receive JWT after creating account

#### How should this be manually tested?

-  clone it 
- run npm i to install dependencies
- run `npm  run start`

#### Any background context you want to provide?

- all fields are required you can not create account with invalid input

#### What are the relevant pivotal tracker stories?

[#169817862](https://www.pivotaltracker.com/story/show/169817862)

#### Screenshots (if appropriate)
Screen Shot 2019-12-19 at 09 18 49

#### Questions:

N/A

### CODE REVIEW

**Comment by shemaeric1234** [comment]
Remove coverage files

File: coverage/base.css [hunk 1]
```diff
@@ -0,0 +1,224 @@
+body, html {
```
Comment (aanchirinah) [inline coverage/base.css]: ### Nitpick

- We don't need this commited to Github

File: coverage/block-navigation.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
+/* eslint-disable */
```
Comment (aanchirinah) [inline coverage/block-navigation.js]: ### Critical

- Include the coverage directory inside `.gitignore`

**Review by aanchirinah — CHANGES_REQUESTED** [review]
Use a NodeJs `.gitignore` template

File: coverage/block-navigation.js (hunk 2 above)
Comment (mujohn26) [reply coverage/block-navigation.js]: yeah I have done

**Review by jo-rdan — APPROVED** [review]
You did good John!

File: src/controllers/userController.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+* class for userControl
+ */
+class userController {
+/**
+   * signup a user and saving user data in the database
+   * @param {Object} req The request object
+   * @param {Object} res The response object
+   * @returns {Object} A user object with selected fields
+   */
+  static async signup(req, res) {
```
Comment (aanchirinah) [inline src/controllers/userController.js]: ### Nitpick

- You are using `Async` but I see no `Await block of code`.

File: src/helpers/responseHandler.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+/**
+* class for responses
+ */
+class response {
+  /**
+   * getting response
```
Comment (aanchirinah) [inline src/helpers/responseHandler.js]: ### Nitpick

- Not descriptive enough

File: src/helpers/responseHandler.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  static errorMessage(req, res, msg, status) {
+    res.status(status).json({
+      status,
+      error: msg,
+
+    });
+  }
+
+  /**
+   * getting response
```
Comment (aanchirinah) [inline src/helpers/responseHandler.js]: ### Nitpick

- Not descriptive enough

File: src/services/UserService.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { user } from '../database/models';
+import Queries from './Queries';
+/**
+* class for responses
```
Comment (aanchirinah) [inline src/services/UserService.js]: ### Nitpick 

- Don't think that's a good description

File: src/tests/user/userData.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+const userData = {
```
Comment (aanchirinah) [inline src/tests/user/userData.js:4]: ### Nitpick

- You can also tryout `faker.js`

**Review by aanchirinah — CHANGES_REQUESTED** [review]
Good job , please attend to requested changes.

File: src/controllers/userController.js (hunk 3 above)
Comment (mujohn26) [reply src/controllers/userController.js]: I solved it

File: src/services/UserService.js (hunk 6 above)
Comment (mujohn26) [reply src/services/UserService.js]: I added a lot more description

File: src/tests/user/userData.js (hunk 7 above)
Comment (mujohn26) [reply src/tests/user/userData.js:4]: I implemented faker

File: src/helpers/responseHandler.js (hunk 4 above)
Comment (mujohn26) [reply src/helpers/responseHandler.js]: I resolved it by adding more descriptive

File: coverage/base.css (hunk 1 above)
Comment (mujohn26) [reply coverage/base.css]: I resolved it

**Review by shemaeric1234 — COMMENTED** [review]
Validation is not working properly and coverage decreased

File: src/controllers/user.controller.js [hunk 8]
```diff
@@ -1,11 +1,133 @@
...
+        phoneNumber,
+        isVerified: false,
+        token
+      };
+      UserServices.CreateUser(NewUser);
+      const data = {
+        token,
+      };
+      const userName = firstName;
+      const view = `<!DOCTYPE html>
```
Comment (dcDalin) [inline src/controllers/user.controller.js]: I think it would be nice to set create email templates in another folder.

**Review by dcDalin — CHANGES_REQUESTED** [review]

File: src/controllers/user.controller.js (hunk 8 above)
Comment (mujohn26) [reply src/controllers/user.controller.js]: Thanks,I solved it !

**Review by shemaeric1234 — APPROVED** [review]

**Pull request merged**